# 1.1 Movies Database

Take the movies dataset and turn it into a single `sqlite` database. It should have one table for each csv file in the movies dataset

In [1]:
import sqlite3
import pandas as pd

In [2]:
df1 = pd.read_csv('data\movies_metadata.csv')
df1 = df1[~df1['budget'].str.contains('jpg')]
df1['budget'] = df1['budget'].astype(int)

C:\Users\pchiu\Documents\Anaconda\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (10) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [3]:
df1.to_csv('new_movies.csv')

In [4]:
conn = sqlite3.connect('data/movies1.db')
c = conn.cursor()

In [5]:
movies_metadata = pd.read_csv('new_movies.csv')
movies_metadata.to_sql('movies_metadata', conn, if_exists='replace', index = False)

C:\Users\pchiu\Documents\Anaconda\lib\site-packages\pandas\core\generic.py:2653: UserWarning: The spaces in these column names will not be changed. In pandas versions < 0.14, spaces were converted to underscores.
  sql.to_sql(


In [6]:
credits = pd.read_csv('data/credits.csv')
credits.to_sql('credits', conn, if_exists='replace', index = False)

In [7]:
keywords = pd.read_csv('data/keywords.csv')
keywords.to_sql('keywords',conn, if_exists='replace', index = False)

In [8]:
links_small = pd.read_csv('data/links_small.csv')
links_small.to_sql('links_small',conn, if_exists='replace', index = False)

In [9]:
rating = pd.read_csv("data/ratings.csv")
rating.to_sql('rating', conn, if_exists='replace', index = False)

In [10]:
ratings_small = pd.read_csv("data/ratings_small.csv")
ratings_small.to_sql('ratings_small', conn, if_exists='replace', index = False)

In [11]:
p2 = pd.read_sql('select original_title from movies_metadata WHERE budget>5000000', conn)
p2

original_title
0                        Toy Story
1                          Jumanji
2                Waiting to Exhale
3                             Heat
4                          Sabrina
...                            ...
5166                       Detroit
5167                        சிவாஜி
5168                       Pattaya
5169  Les Visiteurs: La Révolution
5170     House of the Long Shadows

[5171 rows x 1 columns]

# 1.2 Queries

**1.2.1** Use a single query to pull the original title of movies with a budget above $5m

**1.2.2** Use a query to pull the english-language films with the word `war` in their title

**1.2.3** Left join the average ratings from the `ratings` table onto the `movies_metadata` table, so you can have a relation between budget and rating. Hint: use a subquery.

In [12]:
#1.2.1
p2 = pd.read_sql('select original_title from movies_metadata WHERE budget>5000000', conn)
p2

original_title
0                        Toy Story
1                          Jumanji
2                Waiting to Exhale
3                             Heat
4                          Sabrina
...                            ...
5166                       Detroit
5167                        சிவாஜி
5168                       Pattaya
5169  Les Visiteurs: La Révolution
5170     House of the Long Shadows

[5171 rows x 1 columns]

In [13]:
#1.2.2 (not sure which you wanted)
df2 = pd.read_sql('SELECT original_title FROM clean_movie WHERE original_title LIKE "% war %" ', conn)
df2.head()

original_title
0           The War Room
1        The War at Home
2  The War of the Worlds
3           The War Zone
4   The War of the Roses

In [14]:
#1.2.2 (not sure which you wanted)
df3 = pd.read_sql('SELECT original_title FROM clean_movie WHERE original_title LIKE "%war%" ', conn)
df3.head()

original_title
0                        Star Wars
1               Once Were Warriors
2                          The War
3                     The War Room
4  Snow White and the Seven Dwarfs

In [15]:
#1.2.3 (not sure which you wanted)
df4 = pd.read_sql("""SELECT movieId, movies_metadata.original_title, AVG(rating) ,movies_metadata.budget
FROM ratings_small 
LEFT JOIN movies_metadata 
ON ratings_small.movieId = movies_metadata.id 
GROUP BY movieId""", conn)
df4

movieId        original_title  AVG(rating)     budget
0           1                  None     3.872470        NaN
1           2                 Ariel     3.401869        0.0
2           3  Varjoja paratiisissa     3.161017        0.0
3           4                  None     2.384615        NaN
4           5            Four Rooms     3.267857  4000000.0
...       ...                   ...          ...        ...
9061   161944                  None     5.000000        NaN
9062   162376                  None     4.500000        NaN
9063   162542                  None     5.000000        NaN
9064   162672                  None     3.000000        NaN
9065   163949                  None     5.000000        NaN

[9066 rows x 4 columns]

In [16]:
#1.2.3 (not sure which you wanted)
df5 = pd.read_sql("""SELECT ratings_small.movieId, movies_metadata.original_title, AVG(ratings_small.rating),budget
FROM movies_metadata 
LEFT JOIN ratings_small 
ON movies_metadata.id = ratings_small.movieId 
GROUP BY ratings_small.movieId""", con=conn)
df5

movieId        original_title  AVG(ratings_small.rating)    budget
0          NaN             Toy Story                        NaN  30000000
1          2.0                 Ariel                   3.401869         0
2          3.0  Varjoja paratiisissa                   3.161017         0
3          5.0            Four Rooms                   3.267857   4000000
4          6.0        Judgment Night                   3.884615         0
...        ...                   ...                        ...       ...
2826  140174.0   Rise of the Zombies                   3.900000         0
2827  142507.0                  Exit                   3.000000         0
2828  148652.0   The Eleventh Victim                   2.500000         0
2829  158238.0           Stolen Seas                   3.750000         0
2830  160718.0         I Spy Returns                   4.000000         0

[2831 rows x 4 columns]

# 2. Baseball Database

The [Baseball Database](http://www.seanlahman.com/baseball-archive/statistics/) has an sqlite version. Download it for these exercises.

**2.1** Which player has had the most homeruns?

**2.2** Is there a relation between how many homeruns a player has made in a year and his salary that year? Pull both colums together in a single query



In [17]:
conn1 = sqlite3.connect('data/baseball.db')
c = conn1.cursor()

In [18]:
batting = pd.read_csv('data/baseballdatabank-master/core/Batting.csv')
batting.to_sql('Batting', conn1, if_exists='replace', index = False)

Salaries = pd.read_csv('data/baseballdatabank-master/core/Salaries.csv')
Salaries.to_sql('Salaries', conn1, if_exists='replace', index = False)

In [19]:
bb1 = pd.read_sql("""SELECT SUM(HR) as HR, playerID
FROM Batting
GROUP BY playerID
ORDER BY HR DESC""", conn1)

bb1

HR   playerID
0      762  bondsba01
1      755  aaronha01
2      714   ruthba01
3      696  rodrial01
4      662  pujolal01
...    ...        ...
19893    0  abadijo01
19894    0   abadfe01
19895    0   abadan01
19896    0   aasedo01
19897    0  aardsda01

[19898 rows x 2 columns]

In [21]:
bb2 = pd.read_sql("""SELECT salary, HR, s.yearID, b.playerID
FROM salaries s
JOIN batting b
    ON b.playerID = s.playerID
    AND b.yearID = s.yearID
   
ORDER BY HR DESC
LIMIT 10""", conn1)
bb2

salary  HR  yearID   playerID
0  10300000  73    2001  bondsba01
1   8928354  70    1998  mcgwima01
2   8325000  66    1998   sosasa01
3   9358667  65    1999  mcgwima01
4  12500000  64    2001   sosasa01
5   9000000  63    1999   sosasa01
6    355000  58    2006  howarry01
7   4833333  57    2001  gonzalu01
8  22000000  57    2002  rodrial01
9   7885532  56    1997  griffke02

In [22]:
bb3 = pd.read_sql("""SELECT salary, HR, s.yearID, b.playerID
FROM salaries s
JOIN batting b
    ON b.playerID = s.playerID
    AND b.yearID = s.yearID
   
ORDER BY salary dESC
LIMIT 10""", conn1)
bb3

salary  HR  yearID   playerID
0  33000000  30    2009  rodrial01
1  33000000  30    2010  rodrial01
2  33000000   0    2016  kershcl01
3  32571000   0    2015  kershcl01
4  32000000  16    2011  rodrial01
5  31799030   0    2016  greinza01
6  30000000  18    2012  rodrial01
7  30000000   0    2016  priceda01
8  29000000   7    2013  rodrial01
9  28000000  35    2008  rodrial01

In [23]:
from scipy.stats import linregress

linregress(bb3['HR'], bb3['salary'])

LinregressResult(slope=-18417.428508091336, intercept=31487480.027710043, rvalue=-0.1415555366381403, pvalue=0.696477832635459, stderr=45536.72186282755)

P value (0.696477832635459) is more than 0.05 meaning they are not statistically significant.